### PR TITLE
VBM-182: Feature/refresh token

### PR DESCRIPTION
**Pull request by natata7**

### CODE REVIEW

File: packages/mobile/src/services/http/http.service.ts [hunk 1]
```diff
@@ -83,6 +120,24 @@ class Http {
...
+        tokenValue: refreshToken,
+      };
+      const refreshResponse = await authApi.refreshToken(payload);
+
+      storage.setItem(StorageKey.ACCESS_TOKEN, refreshResponse.accessToken);
+      storage.setItem(StorageKey.REFRESH_TOKEN, refreshResponse.refreshToken);
+
+      if (!refreshResponse.accessToken) {
+        store.dispatch(authActions.logOut());
+      }
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/services/http/http.service.ts:137]: It should go before storage set and I'd recommend throwing a custom error here and then handle it within error handler middleware

File: packages/mobile/src/services/http/http.service.ts [hunk 2]
```diff
@@ -33,14 +41,43 @@ class Http {
...
+
+  private async makeRequest<T = unknown>(
+    url: string,
+    config: { method: HttpMethod; headers: Headers; body: BodyInit_ },
+  ): Promise<T> {
+    const result = await fetch(url, config);
+
+    if (result.status === HttpStatusCode.UNAUTHORIZED) {
+      await this.updateAuthorizationToken();
+      const updatedConfig = this.getRefreshedAuthReqConfig(config);
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/services/http/http.service.ts:59]: This logic seems to belong to `checkStatus`

Comment (natata7) [reply packages/mobile/src/services/http/http.service.ts:59]: it makes sense. but it seems to me that in this case it is not very convenient to pass the config between the initial and the next fetch after updating the token

File: packages/mobile/src/store/middlewares/errors-handler.ts [hunk 3]
```diff
@@ -2,15 +2,22 @@ import { t } from 'i18next';
 import { notification } from '~/services/services';
 import { AppDispatch } from '~/common/types/types';
 import { Middleware, AnyAction } from '@reduxjs/toolkit';
+import { auth as authActions } from '~/store/actions';
+import { ExceptionName } from '@vse-bude/shared';
+import { store } from '../store';
 
 const errorHandler: Middleware =
   () => (next: AppDispatch) => (action: AnyAction) => {
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/store/middlewares/errors-handler.ts:10]:
```suggestion
  ({ dispatch }) => (next: AppDispatch) => (action: AnyAction) => {
```

File: packages/mobile/src/store/middlewares/errors-handler.ts [hunk 4]
```diff
@@ -2,15 +2,22 @@ import { t } from 'i18next';
...
     if (action.error) {
       const {
-        error: { message },
+        error: { name, message },
       } = action;
 
       notification.error(message, t('common:common.ERROR'));
+
+      if (name === ExceptionName.INVALID_CREDENTIALS) {
+        store.dispatch(authActions.logOut());
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/store/middlewares/errors-handler.ts:19]:
```suggestion
       dispatch(authActions.logOut());
```

**Review by AleksandrTermenzhy — APPROVED** [review]
Let's fix middleware and we're good

**Review by VladimirBludov — APPROVED** [review]

**Pull request merged**